### PR TITLE
stream: send nat pinhole packet also if mnat is configured

### PR DESCRIPTION
**Pull request by cspiel1**
Fixes: #2497

### CODE REVIEW

**Comment by cspiel1** [comment]
can we merge this?

**Comment by cspiel1** [comment]
Didn't want to set the milestone. Instead was looking where the milestones can be planned to see when I should make the next release.

**Comment by sreimers** [comment]
Milestone set for May, I think we should wait for feedback by @alfredh

**Comment by alfredh** [comment]
regarding medianat, most of the modules we have should do opening of NAT pinholes.
So there should be no need for explicit packet.

Media NAT modules:

* stun
* turn
* ice
* natpmp
* pcp

Please also note that RTP/RTCP mux is the recommended solution from IETF
regarding NAT keepalives.

Adding more complexity to solve a problem, without understanding the underlying problem,
is wrong IMO. We should try to debug the problem and see if it happens for all medianat types,
or just a few.

**Comment by cspiel1** [comment]
I'll try to figure out the problem with the STUN configuration.

**Comment by cspiel1** [comment]
In our pcap we had different IP addresses for
- stun.linphone.org (51.89.71.129)
- sip.linphone.org (147.135.128.132)

Now nslookup tells 51.89.71.129 for both servers. We'll repeat the test again without the commit in this PR.

**Comment by cspiel1** [comment]
We think that at least for STUN a dummy packet has to be sent to the peer to open the NAT pinhole. Because the stun module will not process the SDP offer, nor send something to the IP:port mentioned in it.

Issues currently are:
- Early video can't pass the NAT of the callee.
- For audio+video calls without early video, the callee may miss the key frame.

File: src/stream.c [hunk 1]
```diff
@@ -1611,7 +1611,8 @@ int stream_open_natpinhole(struct stream *strm)
 	if (!strm)
 		return EINVAL;
 
-	if (strm->pinhole)
+	if (strm->pinhole && (!strm->mnat ||
+			      !str_cmp(strm->mnat->id, "stun")))
```
Comment (alfredh) [inline src/stream.c:1615]: not sure about this. This means that the core gets knowledge about module-names.

I would prefer to delete the STUN module instead, and force ICE usage.

Comment (cspiel1) [reply src/stream.c:1615]: Ah, okay. Is ICE without a TURN server possible?
Does this open the NAT pinhole?

**Comment by alfredh** [comment]
I think this PR is not needed.

Please use ICE instead of STUN. Please also try to understand *why* your setup is not working.

**Pull request closed without merging**